src/sipreg/reg.c

```diff
@@ -223,6 +224,11 @@ static void response_handler(int err, const struct sip_msg *msg, void *arg)
 
 		case 401:
 		case 407:
+			if (reg->ls.failc > 1 && last_scode == msg->scode) {
+				reg->failc++;
+				goto out;
+			}
```

Comment:
After I want to refactor some simple parts, I wonder if this really needed/works, since `sip_request_loops`, does the same for any >=400 scode already:
```c
	else {
		switch (scode) {

		default:
			if (ls->last_scode == scode)
				loop = true;
			/*@fallthrough@*/
		case 401:
		case 407:
		case 491:
			if (++ls->failc >= 16)
				loop = true;
			break;
		}
	}
```
Maybe I miss something. We should add a test case and try to refactor. @maximilianfridrich